packages/dd-trace/src/profiling/ssi-telemetry.js

```diff
@@ -0,0 +1,167 @@
+ * the profiler would, providing insight into how many profiles would've been emitted if SSI enabled
+ * profiling. Note that telemetry is per tracer instance, and each worker thread will have its own instance.
+ */
+class SSITelemetry {
+  constructor ({
+    enablementChoice = getEnablementChoiceFromEnv(),
+    shortLivedThreshold = DEFAULT_SHORT_LIVED_THRESHOLD
+  } = {}) {
+    if (!Object.values(EnablementChoice).includes(enablementChoice)) {
+      throw new Error('Invalid enablement choice')
+    }
+    if (typeof shortLivedThreshold !== 'number' || shortLivedThreshold <= 0) {
+      throw new Error('Short-lived threshold must be a positive number')
+    }
+    this.enablementChoice = enablementChoice
+    this.shortLivedThreshold = shortLivedThreshold
+
+    this.hasSentProfiles = false
+    this.noSpan = true
+  }
+
+  enabled () {
+    return this.enablementChoice !== EnablementChoice.DISABLED
+  }
+
+  start () {
+    if (this.enabled()) {
+      // Used to determine short-livedness of the process. We could use the process start time as the
+      // reference point, but the tracer initialization point is more relevant, as we couldn't be
+      // collecting profiles earlier anyway. The difference is not particularly significant if the
+      // tracer is initialized early in the process lifetime.
+      this.startTime = performance.now()
+
+      this._onSpanCreated = this._onSpanCreated.bind(this)
+      this._onProfileSubmitted = this._onProfileSubmitted.bind(this)
+      this._onMockProfileSubmitted = this._onMockProfileSubmitted.bind(this)
+      this._onAppClosing = this._onAppClosing.bind(this)
+
+      dc.subscribe('dd-trace:span:start', this._onSpanCreated)
+      dc.subscribe('datadog:profiling:profile-submitted', this._onProfileSubmitted)
+      dc.subscribe('datadog:profiling:mock-profile-submitted', this._onMockProfileSubmitted)
+      dc.subscribe('datadog:telemetry:app-closing', this._onAppClosing)
+    }
+  }
+
+  _onSpanCreated () {
+    this.noSpan = false
+    dc.unsubscribe('dd-trace:span:start', this._onSpanCreated)
+  }
+
+  _onProfileSubmitted () {
+    this.hasSentProfiles = true
+    this._incProfileCount()
+  }
+
+  _onMockProfileSubmitted () {
+    this._incProfileCount()
+  }
+
+  _incProfileCount () {
+    this._ensureProfileMetrics()
+    this._profileCount.inc()
+  }
+
+  _ensureProfileMetrics () {
+    const decision = []
+    if (this.noSpan) {
+      decision.push('no_span')
+    }
+    if (performance.now() - this.startTime < this.shortLivedThreshold) {
+      decision.push('short_lived')
+    }
+    if (decision.length === 0) {
+      decision.push('triggered')
+    }
+
+    const tags = [
+      'installation:ssi',
+      `enablement_choice:${enablementChoiceToTagValue(this.enablementChoice)}`,
+      `has_sent_profiles:${this.hasSentProfiles}`,
+      `heuristic_hypothetical_decision:${decision.join()}`
```

Comment:
I don't know if could be a problem that `heuristic_hypothetical_decision` tag value is a concatenation of values, for example if you wanted to know the `short_lived` only metrics.

Reply:
We've been tossing ideas around for this one. All the queries we anticipate are formulated in terms of `heuristic_hypothetical_decision:triggered` or `!heuristic_hypothetical_decision:triggered` so the exact other values aren't terribly significant. It looks like tags support wildcards, so `heuristic_hypothetical_decision:*shortLived` should work.